Create cluster addon failed with

<!--
Hi there,

Thank you for opening an issue. Please note that we try to keep the issue tracker reserved for bug reports and feature requests. For general usage questions, please see: https://www.terraform.io/community.html.

If your issue relates to Terraform, please open it in the terraform's repository at https://github.com/hashicorp/terraform.
-->

### Terraform Version
<!---
Run `terraform -v` to show the version, and paste the result between the ``` marks below.

If you are not running the latest version of Terraform, please try upgrading because your issue may have already been fixed.
-->

```bash
$ terraform version
Terraform v1.5.7
on linux_amd64
+ provider registry.terraform.io/hashicorp/http v3.4.0
+ provider registry.terraform.io/hashicorp/local v2.4.0
+ provider registry.terraform.io/huaweicloud/huaweicloud v1.55.0
```

### Terraform Configuration Files
<!--
Paste the relevant parts of your Terraform configuration between the ``` marks below.

For large Terraform configs, please use a service like Dropbox and share a link to the ZIP file. For security, you can also encrypt the files using our GPG public key.
-->

```hcl

```

### Debug Output
<!--
Full debug output can be obtained by running Terraform with the environment variable `TF_LOG=trace`. Please create a GitHub Gist containing the debug output. Please do _not_ paste the debug output in the issue, since debug output is long.

Debug output may contain sensitive information. Please review it before posting publicly, and if you are concerned feel free to encrypt the files using the HashiCorp security public key.
-->

When I did not set the `values` cause it said that it's a optional parameter

```bash
╷
│ Error: error creating CCE add-on: Bad request with: [POST https://5cf4bce0-4ef5-11ee-a81f-0255ac100100.cce.la-north-2.myhuaweicloud.com/api/v3/addons], request_id: d0b1fb61421ed6661869e941526ead57, error message: {"kind":"Status","apiVersion":"v1","metadata":{},"status":"Failure","code":400,"errorCode":"CCE.03400001","errorMessage":"Invalid request.","error":{"message":"custom values are not defined","code":"SVCSTG.CCE-ADDONMGR.4000002"},"message":"custom values are not defined","reason":"BadRequest"}
│ 
│ 
│   with module.cce.huaweicloud_cce_addon.addons["autoscaler"],
│   on modules/huaweicloud/cce/cluster-addon.tf line 2, in resource "huaweicloud_cce_addon" "addons":
│    2: resource "huaweicloud_cce_addon" "addons" {
│ 
╵
```

When I set `values` with no parameters cause it said that all paremeters of values are optional as well, then got error

```bash
│ Error: Invalid combination of arguments
│ 
│   with module.cce.huaweicloud_cce_addon.addons,
│   on modules/huaweicloud/cce/cluster-addon.tf line 9, in resource "huaweicloud_cce_addon" "addons":
│    9:   values {}
│ 
│ "values.0.basic": one of `values.0.basic,values.0.basic_json` must be specified
╵
╷
│ Error: Invalid combination of arguments
│ 
│   with module.cce.huaweicloud_cce_addon.addons,
│   on modules/huaweicloud/cce/cluster-addon.tf line 9, in resource "huaweicloud_cce_addon" "addons":
│    9:   values {}
│ 
│ "values.0.basic_json": one of `values.0.basic,values.0.basic_json` must be specified
╵
```

When I set `basic_json` then got carashed
```bash
  values {
    basic_json = ""
  }
```
crashed
```bash
╷
│ Error: Request cancelled
│ 
│   with module.cce.huaweicloud_cce_addon.addons["autoscaler"],
│   on modules/huaweicloud/cce/cluster-addon.tf line 2, in resource "huaweicloud_cce_addon" "addons":
│    2: resource "huaweicloud_cce_addon" "addons" {
│ 
│ The plugin.(*GRPCProvider).ApplyResourceChange request was cancelled.
╵
╷
│ Error: Plugin did not respond

│ The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may contain more
│ details.
╵

Stack trace from the terraform-provider-huaweicloud_v1.55.0 plugin:

panic: interface conversion: interface {} is nil, not map[string]interface {}

goroutine 53 [running]:
github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/cce.buildAddonValues(0xc0011e2900?)
        github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/cce/resource_huaweicloud_cce_addon.go:140 +0x585
github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/cce.resourceAddonCreate({0x2504458, 0xc000af1d50}, 0x0?, {0x215ef20?, 0xc0011e2900?})
        github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/cce/resource_huaweicloud_cce_addon.go:182 +0x1d9
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).create(0xc00032b260, {0x25043b0, 0xc000b6fad0}, 0xd?, {0x215ef20, 0xc0011e2900})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.0/helper/schema/resource.go:707 +0x11b
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc00032b260, {0x25043b0, 0xc000b6fad0}, 0xc000c004e0, 0xc0011df600, {0x215ef20, 0xc0011e2900})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.0/helper/schema/resource.go:837 +0xa69
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc000010138, {0x25043b0?, 0xc000b6f9e0?}, 0xc000c5c9b0)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.24.0/helper/schema/grpc_provider.go:1021 +0xdbc
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0xc000943ea0, {0x25043b0?, 0xc000b6f1d0?}, 0xc000af1570)
        github.com/hashicorp/terraform-plugin-go@v0.14.0/tfprotov5/tf5server/server.go:818 +0x56a
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0x20bcbc0?, 0xc000943ea0}, {0x25043b0, 0xc000b6f1d0}, 0xc000af1500, 0x0)
        github.com/hashicorp/terraform-plugin-go@v0.14.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:385 +0x169
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0002b81e0, {0x2508e10, 0xc000acf1e0}, 0xc000b71d40, 0xc000a89740, 0x35ca3e0, 0x0)
        google.golang.org/grpc@v1.53.0/server.go:1336 +0xd67
google.golang.org/grpc.(*Server).handleStream(0xc0002b81e0, {0x2508e10, 0xc000acf1e0}, 0xc000b71d40, 0x0)
        google.golang.org/grpc@v1.53.0/server.go:1704 +0x9e7
google.golang.org/grpc.(*Server).serveStreams.func1.2()
        google.golang.org/grpc@v1.53.0/server.go:965 +0x8d
created by google.golang.org/grpc.(*Server).serveStreams.func1 in goroutine 13
        google.golang.org/grpc@v1.53.0/server.go:963 +0x246

Error: The terraform-provider-huaweicloud_v1.55.0 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```

### Crash Output
<!--
If the console output indicates that Terraform crashed, please share a link to a GitHub Gist containing the output of the `crash.log` file.
-->

### Expected Behavior
<!--
What should have happened?
-->

### Actual Behavior
<!--
What actually happened?
-->

### Steps to Reproduce
<!--
Please list the full steps required to reproduce the issue, for example:
1. `terraform init`
2. `terraform apply`
-->

### Additional Context
<!--
Are there anything atypical about your situation that we should know? For example: is Terraform running in a wrapper script or in a CI system? Are you passing any unusual command line options or environment variables to opt-in to non-default behavior?
-->

### References
<!--
Are there any other GitHub issues (open or closed) or Pull Requests that should be linked here? For example:

- #6017

-->

--- a/huaweicloud/services/cce/resource_huaweicloud_cce_addon.go
+++ b/huaweicloud/services/cce/resource_huaweicloud_cce_addon.go
@@ -132,7 +132,7 @@ func ResourceAddon() *schema.Resource {
 
 func buildAddonValues(d *schema.ResourceData) (basic, custom, flavor map[string]interface{}, err error) {
 	values := d.Get("values").([]interface{})
-	if len(values) == 0 {
+	if len(values) == 0 || values[0] == nil {
 		basic = make(map[string]interface{})
 		return
 	}

